connectgateway/get_namespace.py

```diff
@@ -0,0 +1,106 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# [START connectgateway_get_namespace]
+import os
+import sys
+
+from google.api_core import exceptions
+from google.auth.transport import requests
+from google.cloud.gkeconnect import gateway_v1
+from google.oauth2 import service_account
+from kubernetes import client
+
+SCOPES = ['https://www.googleapis.com/auth/cloud-platform']
+
+
+def get_gateway_url(membership_name: str, location: str) -> str:
+    """Fetches the GKE Connect Gateway URL for the specified membership."""
+    try:
+        client_options = {}
+        if location != "global":
+            # If the location is not global, the endpoint needs to be set to the regional endpoint.
+            regional_endpoint = f"{location}-connectgateway.googleapis.com"
+            client_options = {"api_endpoint": regional_endpoint}
+        gateway_client = gateway_v1.GatewayControlClient(client_options=client_options)
+        request = gateway_v1.GenerateCredentialsRequest()
+        request.name = membership_name
+        response = gateway_client.generate_credentials(request=request)
+        print(f'GKE Connect Gateway Endpoint: {response.endpoint}')
+        if not response.endpoint:
+            print("Error: GKE Connect Gateway Endpoint is empty.")
+            sys.exit(1)
+        return response.endpoint
+    except exceptions.NotFound as e:
+        print(f'Membership not found: {e}')
+        sys.exit(1)
+    except Exception as e:
+        print(f'Error fetching GKE Connect Gateway URL: {e}')
+        sys.exit(1)
+
+
+def configure_kubernetes_client(gateway_url: str, service_account_key_path: str) -> client.CoreV1Api:
+    """Configures the Kubernetes client with the GKE Connect Gateway URL and credentials."""
+
+    configuration = client.Configuration()
+
+    # Configure the API client with the custom host.
+    configuration.host = gateway_url
+
+    # Set the authentication header with the GCP OAuth token
+    if not service_account_key_path:
+        print("Error: SERVICE_ACCOUNT_KEY_PATH environment variable not set.")
+        sys.exit(1)
+    try:
+        credentials = service_account.Credentials.from_service_account_file(
+            service_account_key_path, scopes=SCOPES
+        )
+    except FileNotFoundError:
+        print(f"Error: Service account key file not found at {service_account_key_path}")
+        sys.exit(1)
+    except Exception as e:
+        print(f"Error loading service account credentials: {e}")
+        sys.exit(1)
```

Comment:
issue(blocking): we do not support service account key based authentication, for either sample testing, nor samples where other authentication methods exist. Can this sample work using ADC?